emcc.py

```diff
@@ -682,7 +682,10 @@ def uniquename(name):
       else:
         raise
 
-  temp_dir = tempfile.mkdtemp(dir=temp_root)
+  if not DEBUG:
+    temp_dir = tempfile.mkdtemp(dir=temp_root)
+  else:
+    temp_dir = shared.get_emscripten_temp_dir()
```

Comment:
Actually I realized why this might not work, in general.

If there is more than one invocation of emcc (i.e. -j N) then this approach can cause compilation failures.  With the old approach files are copied from the unique temp dir to the common temp dir, which means they will clobber each other but it will be harmless.

With the approach intermediate files that are used as input will clobber each other, which can cause failures.

To work around this we could suggest and/or enforce that under EMCC_DEBUG only one compilation process can occur at a given time?   This might not be a bad idea... since the log messages from the different compiler processes will already be impossible to read if they are interleaved.

Reply:
I think you're right that we can simplify this, I opened #7727.

It's true that with EMCC_DEBUG multiple processes can be a problem. I thought that was always the case, but maybe that's not quite right - good point that we should just ensure one process in that case.